bootstrap/mod/src/main/java/org/geysermc/geyser/platform/mod/GeyserModBootstrap.java

```diff
@@ -214,6 +209,17 @@ public boolean testFloodgatePluginPresent() {
         return this.platform.testFloodgatePluginPresent(this);
     }
 
+    private Path floodgateKeyPath;
+
+    public void loadFloodgate(Path floodgateDataFolder) {
+        floodgateKeyPath = FloodgateKeyLoader.getKeyPath(geyserConfig, floodgateDataFolder, dataFolder, geyserLogger);
+    }
+
+    @Override
+    public Path getFloodgateKeyPath() {
+        return floodgateKeyPath;
+    }
```

Comment:
why isn't the FloodgateKeyPath just calling FloodgateKeyLoader, like other implementations? and why is testFloodgatePluginPresent the one responsible for calling this loadFloodgate method (seems like it was like that before tho)?

Reply:
As long as it works, I'm not too concerned, as it will be yeeted sometime in the short term.